tests/plugins/link/anchor.js

```diff
@@ -79,6 +79,46 @@
 					assert.areSame( anchor, dialog.getModel( editor ) );
 				} );
 			} );
+		},
+
+		// (#3863)
+		'test preserve duplicate anchors after editing word with custom style': function() {
+			var editor = this.editor,
+				bot = this.editorBot,
+				html = '<p><a id="test" name="test"><strong>text</strong></a></p>';
+
+			bot.setData( html, function() {
+				var range = editor.createRange(),
+					anchor = editor.editable().findOne( '#test' ),
+					textNode = anchor.getChild( 0 );
+
+				range.selectNodeContents( textNode );
+				range.select();
+
+				bot.dialog( 'anchor', function( dialog ) {
+					dialog.setValueOf( 'info', 'txtName', 'duplicate-test' );
+					dialog.getButton( 'ok' ).click();
+
+					assert.areSame( anchor, dialog.getModel( editor ) );
+				} );
+
+			} );
+		},
+
+		// (#4728)
+		'test prevent duplicate anchors in selected multi words with styles': function() {
```

Comment:
```suggestion
		'test prevent duplicated anchors after editing multiple words with styles': function() {
```

Maybe something like :point\_up: